### DISCUSSION TITLE
Play Streamlink stream

### DISCUSSION THREAD

**Discussion by k2helix** [videojs/video.js#8207] [Q&A] [answered]
Streamlink is a tool that is able to play streams from sites such as Twitch or others, and can make that stream available in some port by using the option --player-external-http, so that the ip and port are logged when using it (for example: 127.0.0.1:12091)

Is there a way to pipe the stream Streamlink provides when using the option --player-external-http to video.js?

Just using ip:port as the video src does not seem to work.

**Comment by campones** [comment]
it's an interesting idea

**Comment by campones** [comment]
streamlink team (well currently only one is on it.. ) doesn't want to restream , i m pretty confident it would have been easy for them to implement the option to output to an rtmp server but.. .  so what you do is pipe to stdout the streamlink input and then use ffmpeg to grab the input and restream where you want 
if you don't want/know to compile ffmpeg, you can use a binary, works well too

**Reply by k2helix** [reply]
I already have ffmpeg binaries. So what args do I need to pass? Let's say I use _streamlink https://twitch.tv/someone best_ , what would I need to do with ffmpeg and what would I pass to video.js src?

**Reply by campones** [reply]
well obviously you ll need an rtmp server.   
 streamlink > stdout > ffmpeg > rtmp server  , then you will have a dash/m3u8 stream input to add as a player source. 

`streamlink https://twitch.tv/someone best -O | ffmpeg -re -i pipe:0 -c:v copy -c:a libfdk_aac -ar 44100 -f flv "rtmp://ip/app/streamname"`

**Reply by campones** [reply]
you could try this thing I discovered recently called nimble streaming server, I think it's kinda based on wowza but "free" except you need a panel to control it, which is paid but after setting everything with the panel, it can work without .

**Reply by k2helix** [reply]
I thought this some time ago too. I read that I need a tool like nginx to create the RTMP server, but do you know of an alternative? To be honest I did not do it because I have never used it.

**Reply by campones** [reply]
well I just told you about the alternative.  setting up an nginx rtmp server is a bit more complex but possibly offer a lot more alternatives. there are unlimited options with nginx if you know how to config it..  but yea it's tough at the beginning

**Reply by campones** [reply]
it really depends of what you know or not. if you host websites with nginx and you compile nginx etc, then you are good to go

**Reply by campones** [reply]
I also recently learnt there was a good fork of the nginx rtmp module (that is not maintained for a very long time now) https://github.com/winshining/nginx-http-flv-module     that will be a good start.  I actually installed this on a vmware and just copied my production server config and it works out of the box

**Reply by k2helix** [reply]
I have been able to start a working rtmp server to which I send the stream with streamlink and ffmpeg. However, in order for video.js to play it, it needs flash which is not something I can provide it. Any ideas?

**Reply by campones** [reply]
that's not possible since flash is obsolete .  your rtmp server has to be setup in a way it transform rtmp input into hls or dash streams output

**Reply by k2helix** [reply, marked as answer]
It's been some time but I was able to resolve this some months ago by using [node-rtmp-hls](https://github.com/johndavedecano/node-rtmp-hls) instead of nginx (with nginx the stream video was corrupted somehow). 

I also made a script to do it automatically which I uploaded [here](https://github.com/k2helix/twitch-local-player).
